### PR TITLE
Content API Models - upgrading to Circe 0.6.1.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**

### CODE REVIEW

File: json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala [hunk 1]
```diff
@@ -33,89 +33,13 @@ object CirceDecoders {
...
         case "true" => Some(true)
         case "false" => Some(false)
         case _ => None
       })
-      Xor.fromOption(fromBooleanOrString, ifNone = DecodingFailure("Boolean", c.history))
+      maybeFromBooleanOrString.map(Right(_)).getOrElse(Left(DecodingFailure("Boolean", c.history)))
     }
   }
 
-  // The following implicits technically shouldn't be necessary
```
Comment (LATaylor-guardian) [inline json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala]: With the upgrade, this now compiles without the need to declare these.

**Comment by LATaylor-guardian** [comment]
Will deploy after Christmas.

**Comment by LATaylor-guardian** [comment]
@gidsg @tomrf1

**Comment by LATaylor-guardian** [comment]
Will merge after Christmas.

File: json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package com.gu.contentapi.json
 
 import io.circe._
-import cats.data.Xor
 import com.gu.contentatom.thrift.{Atom, AtomData}
 import com.gu.contentapi.circe.CirceScroogeMacros._
 import com.gu.contentapi.client.model.v1._
 import org.joda.time.format.ISODateTimeFormat
+import cats.syntax.either._
```
Comment (tomrf1) [inline json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala]: 💯

**Comment by tomrf1** [comment]
👍

File: json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala [hunk 3]
```diff
@@ -16,8 +16,8 @@ object CirceDecoders {
   implicit final val decodeString: Decoder[String] = new Decoder[String] {
     final def apply(c: HCursor): Decoder.Result[String] = {
       val focus = c.focus
-      val fromStringOrLong = focus.asString.orElse(focus.asNumber.flatMap(_.toLong.map(_.toString)))
-      Xor.fromOption(fromStringOrLong, ifNone = DecodingFailure("String", c.history))
+      val maybeFromStringOrLong = focus.asString.orElse(focus.asNumber.flatMap(_.toLong.map(_.toString)))
+      maybeFromStringOrLong.map(Right(_)).getOrElse(Left(DecodingFailure("String", c.history)))
```
Comment (philwills) [inline json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala]: There is a fromOption provided as part of the cats syntax (possibly also in 2.12 stdlib).

Comment (LATaylor-guardian) [inline json/src/main/scala/com/gu/contentapi/json/CirceDecoders.scala]: Thanks, missed that.

**Comment by janua** [comment]
❤️

**Pull request merged**